File: subArray.py
def subArray(numbers, sum):
	for i, value in enumerate(numbers):
		for j in range(i-1, -1, -1):
			value += numbers[j]
			if value == sum:
				return str(j+1) + " " + str(i+1)
	return -1

for x in range(int(input())):
	numberCount, sum = [ int(i) for i in input().split() ]
	numbers = [ int(i) for i in input().split() ]
	print(subArray(numbers, sum))

File: subArray.js
function maximumSum(subArray, number) {
	for (var i = 0; i < subArray.length; i++) {
		var sum = subArray[i];
		for (var j = i-1; j >= 0; j--) {
			sum += subArray[j];
			if (sum == number) return [j+1, i+1];
		}
	}
	return -1
}

console.log(maximumSum([1,2,3,7,5], 12));
